### ISSUE TITLE
Craft 4 support

### ISSUE THREAD

**Issue by acalvino4** [ethercreative/tags#26]
Are there plans to keep this updated for Craft 4?

**Comment by alexjcollins** [comment]
@acalvino4 We'd certainly like to. We just need to determine the work involved so we can schedule accordingly.

I'll leave this open whilst we figure it out.

**Comment by davidwebca** [comment]
Craft core plans to remove Tags and Categories altogether eventually and give Entries UI capabilities that would enable them to be used and displayed as such. https://github.com/craftcms/cms/issues/820

That being said, it doesn't seem like they've moved much forward to adding that in 4.0 since the only core change they've made is to add draft functionalities to categories.

Our website uses tags pretty heavily so I'll try to see what would need to be done to add the compatibility for Craft 4.0 and I'll make a pull request.

**Comment by mmikkel** [comment]
The earliest Tags and Categories could be removed is 5.0, as it would be a major breaking change. In the interim, I'd love to see this plugin upgraded for Craft 4 compatibility.

**Comment by wmdhosting** [comment]
+1

**Comment by pixelbeatcreative** [comment]
+1

**Comment by arifje** [comment]
+1

**Comment by jasonmorehead** [comment]
+1

**Comment by davidwebca** [comment]
Just chiming in for all those +1s: I've been using my fork for a few weeks without any issue so far. So, for those who would need it right away while Ether reviews the pull request, you can pull my fork by changing your composer file like below.

```json
{
  "require": {
    "ether/tags": "dev-master"
  },
  "repositories": [
    { "type": "git", "url": "https://github.com/davidwebca/craft-tags" }
  ]
}
```

Of course, this would be a temporary measure as I can't guarantee that my fork will stay up (I might delete my fork if and when the pull request gets approved). Make sure to update your file later if you do this!

**Comment by swissmexxa** [comment]
+1

**Comment by viesrood** [comment]
+1

**Comment by CreateSean** [comment]
+1

**Comment by ricricucit** [comment]
Curiosity: does @brandonkelly, personally have an opinion on this topic?

**Comment by johndwells** [comment]
+1

**Comment by wmdhosting** [comment]
can you review  PR:.   for craft 4

**Comment by brandonkelly** [comment]
Last week we announced that tags, categories, and global sets are going to be deprecated in Craft 4.4, and fully removed in Craft 5, in favor of entries (craftcms/cms#9781). And in 4.4 we will be adding a CLI tool that can be used to migrate existing tag groups, category groups, and global sets over to entries. We’ll also be making a couple improvements to Entries fields so they can replicate Categories and Tags fields’ UI (craftcms/cms#8522 + https://github.com/craftcms/cms#11379).

So in other words: this plugin won’t be needed anymore :)

**Comment by rob-baker-ar** [comment]
This is great news. Not sure what the timeline is on Craft 5 but I'm guessing it's far enough away that some people may still need this.

**Comment by wmdhosting** [comment]
as much i understand in 4.4  will be avaible.   in 5 just removed completly..
But yes as someone allready posted PR:: for 4.. not much work to make live

**Comment by vebjorn** [comment]
> Last week we announced that tags, categories, and global sets are going to be deprecated in Craft 4.4, and fully removed in Craft 5, in favor of entries).
> 

We need clarification on this then. Does deprecation mean that tags, categories and globals will be replaced with entries in 4.4? Or will deprecation here mean nothing else but a forewarning for Craft 5?

Q4 2023 is quite a while a head. A lot of us can not wait for over a year for something like this. But if 4.4 is a couple of days or weeks away, waiting might be something to consider

**Comment by brandonkelly** [comment]
> We need clarification on this then. Does deprecation mean that tags, categories and globals will be replaced with entries in 4.4? Or will deprecation here mean nothing else but a forewarning for Craft 5?

No, deprecated means those element types will still exist, but we no longer recommend you use them going forward. It doesn’t make sense to deprecate them before that because entries aren’t yet a viable alternative. They will be in 4.4 though.

> Q4 2023 is quite a while a head. A lot of us can not wait for over a year for something like this. But if 4.4 is a couple of days or weeks away, waiting might be something to consider

No need to wait; nothing is changing for the time being. Categories, tags, and global sets are still recommended until 4.4. The CLI migration tool will help you move to entries when the time is right. (Could totally just be something you put off until you’re getting ready to upgrade to Craft 5 – if it ain’t broke don’t fix it.)

> That's an ambitious goal when the official roadmap for Craft says "Convert categories, tags, and global sets into entries" is planned for 5.x ..... there is a massive lack of info, and massive gap between what they say, and what is official info here.

It’s a transition, but 5.0 will deliver the last nail in the coffin, so that’s why it’s marked as a 5.x feature.

**Comment by brandonkelly** [comment]
Yeah fair.

`“Relate ancestors” setting for Entries/Categories fields` was already present for 4.x.

Just added `“Create &lt;element type&gt;” button for relational fields` and `Category/tag/global set → entry conversion tool`, which were missing.

**Comment by theAdhocracy** [comment]
@brandonkelly Are you also looking at porting across the Tags UI? Specifically, the ability to quickly search, select, and create "tags" (I guess that should now just read "entries") all from a single input element? And then double-click to open up an editor where further fields can be filled in?

We're currently stuck on Craft 3 as we need the speed of being able to quickly add new tags that this provides. We've also found that the pop-out list of Categories often takes a long time to load, whereas Tag autocomplete seems to be nearly instantaneous. As a result, we've been slowly migrating almost all Categories to Tags, but I'm now wary of that strategy if we're going to lose the UI in the near future?

**Comment by brandonkelly** [comment]
@theAdhocracy Yeah inline entry creation will be possible from Entries fields, similar to how Tags fields work.

**Comment by smockensturm** [comment]
I wouldn't mind a state of the state here. I don't see anything (as of Craft 4.5.9) nearly replicating the sweet UI of searching, adding and creating tags on the fly that this plugin affords.

I dig this plugin for that alone. I don't want something **similar** or something functionally the same. I want this.

So what's going on? Wil this plugin be Craft 4 compatible? Will @brandonkelly replicate a sweet tagging UI exactly?

**Comment by theAdhocracy** [comment]
@smockensturm Just to be clear, the Tags plugin is available for Craft 4 and it works exactly like Craft 3. We're running it at the moment with no issues. You just need to keep your Tags as, well, Tags, rather than go through the entrification process with them.

As for Craft 5, time will tell. I wasn't able to make dot.all and the video of the keynote is not clear on how the UI functions, but it looks like there will be a variety of options available. I'm in the same boat though; we need the **same** UI experience. So fingers crossed 🤞

**Comment by smockensturm** [comment]
>Just to be clear, the Tags plugin is available for Craft 4 and it works exactly like Craft 3.

Yeah, I suppose. Thanks for the rundown.

<img width="538" alt="Screen Shot 2023-11-17 at 10 21 27 AM" src="https://github.com/ethercreative/tags/assets/6586518/499aaf46-4c00-4a62-a5cc-83df414b2a0a">

**Comment by jonnykates** [comment]
@smockensturm In your previous screenshot, you've got two 'delete' options. We're encountering this problem too whilst using the Tags plugin on Craft 4 (without having entrified our tags). Did you work out a solution for this? I guess one of the 'delete' options is Craft's native delete whereas the other is the Tag plugin handler (which opens a modal giving the user the change to replace etc.).

**Comment by smockensturm** [comment]
@jonnykates No. This plugin apparently is a bust. Because 'entrification' is the way forward.

https://craftcms.com/docs/5.x/reference/element-types/tags.html

But we stiil don't know what @brandonkelly plans for the sweet inline tags UI.

https://craftcms.com/docs/5.x/reference/element-types/tags.html

https://github.com/ethercreative/tags/issues/26#issuecomment-1280858790

**Comment by JayBox325** [comment]
This is still relevant and would be a very helpful plugin.
